atlasdb-config/src/test/java/com/palantir/atlasdb/factory/TransactionManagersTest.java

```diff
@@ -155,6 +178,73 @@ public void userAgentsPresentOnRequestsWithLeaderBlockConfigured() throws IOExce
+                .quorumSize(1)
+                .build()));
+
+        TransactionManagers.LockAndTimestampServices lockAndTimestampServices =
+                TransactionManagers.createLockAndTimestampServices(
+                        config,
+                        environment,
+                        LockServiceImpl::create,
+                        InMemoryTimestampService::new,
+                        invalidator,
+                        USER_AGENT);
+        availableServer.verify(getRequestedFor(urlMatching(LEADER_UUID_PATH)));
+
+        lockAndTimestampServices.time().getFreshTimestamp();
+        lockAndTimestampServices.lock().currentTimeMillis();
+
+        availableServer.verify(postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader(USER_AGENT_HEADER, WireMock.equalTo(USER_AGENT)));
+        availableServer.verify(postRequestedFor(urlMatching(LOCK_PATH))
+                .withHeader(USER_AGENT_HEADER, WireMock.equalTo(USER_AGENT)));
+    }
+
+    @Test
+    public void remoteCallsElidedIfTalkingToLocalServer() throws IOException, InterruptedException {
+        doAnswer(invocation -> {
+            // Configure our server to reply with the same server ID as the registered PingableLeader.
+            PingableLeader localPingableLeader = invocation.getArgumentAt(0, PingableLeader.class);
+            availableServer.stubFor(LEADER_UUID_MAPPING.willReturn(aResponse()
+                    .withStatus(200)
+                    .withBody(("\"" + localPingableLeader.getUUID().toString() + "\"").getBytes())));
+            return null;
+        }).when(environment).register(isA(PingableLeader.class));
+        when(config.leader()).thenReturn(Optional.of(ImmutableLeaderConfig.builder()
+                .localServer(getUriForPort(availablePort))
+                .addLeaders(getUriForPort(availablePort))
+                .acceptorLogDir(temporaryFolder.newFolder())
+                .learnerLogDir(temporaryFolder.newFolder())
+                .quorumSize(1)
+                .build()));
+
+        TransactionManagers.LockAndTimestampServices lockAndTimestampServices =
+                TransactionManagers.createLockAndTimestampServices(
+                        config,
+                        environment,
+                        LockServiceImpl::create,
+                        InMemoryTimestampService::new,
+                        invalidator,
+                        USER_AGENT);
+        availableServer.verify(getRequestedFor(urlMatching(LEADER_UUID_PATH)));
+
+        lockAndTimestampServices.time().getFreshTimestamp();
+        lockAndTimestampServices.lock().currentTimeMillis();
+
+        availableServer.verify(0, postRequestedFor(urlMatching(TIMESTAMP_PATH))
+                .withHeader(USER_AGENT_HEADER, WireMock.equalTo(USER_AGENT)));
+        availableServer.verify(0, postRequestedFor(urlMatching(LOCK_PATH))
+                .withHeader(USER_AGENT_HEADER, WireMock.equalTo(USER_AGENT)));
+    }
+
```

Comment:
In general should there be a test that it still uses the remote service if the service gives us a UUID that is not our local PingableLeader's UUID?

Alternatively I'm open to just deciding that we have undefined behaviour in those circumstances (that happens if you set a 1 node leader block with `ourLeader` and `leaders as NOT ourselves)